docs/contributor/arch/027-switch-helm-packaging-from-kustomize.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Switch Kustomize Packaging to Helm
+status: Accepted
+date: 2025-08-21
+---
+
+# 26. Switch Kustomize Packaging to Helm
+
+## Context
+
+For deploying the module across multiple environments, templating capabilities are frequently required. While Kustomize was initially chosen due to its integration with kubebuilder (which generates artifacts using Kustomize), its role in our setup has become minimal. At this point, Kustomize is only used for CRD generation.
+
+Since these CRDs are unaffected by the deployment layer, we now have the opportunity to switch the main deployment mechanism to Helm, which is the de-facto standard for Kubernetes packaging and offers significantly more powerful templating and configuration features. The proposal is therefore to convert the existing `config/` folder into a Helm chart, and adapt the deployment and testing tooling accordingly.
+
+## Proposal
+
+The proposal is to switch the deployment packaging from Kustomize to Helm. This involves the following steps:
+1. Convert the existing `config/` folder into a Helm chart, maintaining the same structure and functionality.
+2. Adapt the deployment scripts and CI/CD pipelines to use Helm for deploying the module.
+3. Update the documentation to reflect the changes in deployment procedures.
+4. Ensure that all existing functionality is preserved and that the new Helm-based deployment is tested thoroughly.
+5. Maintain the CRD generation using ControllerGen, as it is not affected by the deployment layer.
+
+### CRD Manifests
+The Helm CRDs are treated as a special kind of object. They are installed before the rest of the chart, however, they have some limitations:
```

Comment:
```suggestion
The Helm CRDs are treated as a special kind of object. They are installed before the rest of the chart. However, they have some limitations:
```
What's the relationship between these sentences? They stand right next to each other but how does "they're special" relate to "they're installed before others" and to "they have limitations"? Is there anything that connects these statements?